test/core.vwtest.json

```diff
@@ -5123,5 +5123,35 @@
     "input_files": [
       "train-sets/cb_l_namespace.txt"
     ]
+  },
+  {
+    "id": 399,
+    "desc": "test multi d -- based on test 316 to share same metrics of the complete file dsjson_cb.json (which is split into two here)",
+    "vw_command": "-d train-sets/dsjson_cb_part1.json -d train-sets/dsjson_cb_part2.json --dsjson --cb_explore_adf --epsilon 0.2 --quadratic GT --extra_metrics metrics_time.json",
+    "diff_files": {
+      "metrics_time.json": "test-sets/ref/metrics_time.json",
+      "stdout": "train-sets/ref/metrics_time.stdout",
+      "stderr": "train-sets/ref/multid.stderr"
+    },
+    "input_files": [
+      "train-sets/dsjson_cb_part1.json",
+      "train-sets/dsjson_cb_part2.json"
+    ]
+  },
+  {
+    "id": 400,
+    "desc": "test multi d positional -- based on test 316 to share same metrics of the complete file dsjson_cb.json (which is split into two here)",
+    "vw_command": "--dsjson --cb_explore_adf --epsilon 0.2 --quadratic GT --extra_metrics metrics_time.json train-sets/dsjson_cb_part1.json train-sets/dsjson_cb_part2.json",
```

Comment:
do we want to allow multiple positional arguments for input file? even single one is constant source of problems when someone accidentally adding wrong options. Multiple positional arguments could be source of problems that could be hard to notice (like every wrong commandline option will be treated as input file?). Maybe we should explicitly deprecate that?

Reply:
100% agree. We should not be expanding the positional input.

Reply:
And probably the same about mixed mode (like one input is positional, another - explicit)?

Reply:
yes, but the log should provide help to decipher it. See test 404 for a weird mixed mode side effect in ordering.

Reply:
hm, it still seems too scary to be honest - basically we are introducing a lot of ambiguity in parsing - for example if you have file with the same name as valid command line argument: let's say you have files with names "-q" and "sa" - then what command line "--cb_explore_adf -q sa -d file.txt" is supposed to do?
I understand that it is extreme corner case, but that change is increasing their amount substantially.